### ISSUE TITLE
Handling of nullable entity properties

### ISSUE THREAD

**Issue by kgoncharov** [mikro-orm/mikro-orm#132]
**Describe the bug**
I have an entity with some optional fields
```
@Entity()
export class CorporateUser extends Auditable {
  @Property()
  email: string;

  @Property()
  username: string;

  @Property({ nullable: true })
  firstName?: string;

  @Property({ nullable: true })
  lastName?: string;

  @OneToOne({ owner: true })
  credentials: Credentials;

  @ManyToOne()
  organization: Organization;

  constructor(
    props: Omit<CorporateUser, keyof IEntity | keyof Auditable>,
  ) {
    super();
    this.email = props.email;
    this.username = props.username;

    this.firstName = props.firstName;
    this.lastName = props.lastName;

    this.credentials = props.credentials;
    this.organization = props.organization;
  }
}
``` 
When I'm trying to persist entity w/o `firstName` and `lastName` I got the exception below:
**Stack trace**
```
[2019-09-19 14:06:08.492 +0000] ERROR (2032 on kgoncharov-linux): ExceptionsHandler Trying to set CorporateUser.firstName of type 'string' to 'undefined' of type 'undefined' ValidationError: Trying to set CorporateUser.firstName of type 'string' to 'undefined' of type 'undefined'
    at Function.fromWrongPropertyType (/home/kgoncharov/Projects/ioXt/haxlab-be/node_modules/mikro-orm/dist/utils/ValidationError.js:18:16)
    at EntityValidator.validateProperty (/home/kgoncharov/Projects/ioXt/haxlab-be/node_modules/mikro-orm/dist/entity/EntityValidator.js:36:43)
    at /home/kgoncharov/Projects/ioXt/haxlab-be/node_modules/mikro-orm/dist/entity/EntityValidator.js:21:49
    at Array.forEach (<anonymous>)
    at EntityValidator.validate (/home/kgoncharov/Projects/ioXt/haxlab-be/node_modules/mikro-orm/dist/entity/EntityValidator.js:16:30)
    at ChangeSetComputer.computeChangeSet (/home/kgoncharov/Projects/ioXt/haxlab-be/node_modules/mikro-orm/dist/unit-of-work/ChangeSetComputer.js:21:24)
    at UnitOfWork.findNewEntities (/home/kgoncharov/Projects/ioXt/haxlab-be/node_modules/mikro-orm/dist/unit-of-work/UnitOfWork.js:151:50)
    at UnitOfWork.computeChangeSets (/home/kgoncharov/Projects/ioXt/haxlab-be/node_modules/mikro-orm/dist/unit-of-work/UnitOfWork.js:119:18)
    at UnitOfWork.commit (/home/kgoncharov/Projects/ioXt/haxlab-be/node_modules/mikro-orm/dist/unit-of-work/UnitOfWork.js:76:14)
    at EntityManager.flush (/home/kgoncharov/Projects/ioXt/haxlab-be/node_modules/mikro-orm/dist/EntityManager.js:219:36)
```

What is the right way to:
1. Set nullable properties in `Entity`
2. To handle nulls in database in MikroORM

**Comment by B4nan** [comment]
Hmmm this is currently not supported, but not on purpose... Thanks for the report, will fix this later today. 

As a workaround you can allow `null` in your props:

```typescript
  @Property({ nullable: true })
  firstName: string | null;
```

Then validation should pass.

**Comment by kgoncharov** [comment]
Thanks for clarification @B4nan!
Unfortunately, allowing nulls in entity properties breaks the whole flow in app. 
After your fix, setting of `undefined` to property will lead to `table_name.property = null` in db, right?
Also, when are you planning to release new version with this fix?

**Comment by B4nan** [comment]
Yes, will release it right after its implemented as v2.7.8. And yes, it will set the value to `null` in database.

**Comment by B4nan** [comment]
Looking into this, its pretty easy fix, but there is one caveat. All drivers currently return `null` when reading such property from database. It is fine unless you would check specifically for `undefined` via strict equality:

```typescript
const user = await orm.em.findOne(User, 1);
user.firstName = undefined;
console.log(user.firstName); // undefined
await orm.em.flush();
orm.em.clear();

// reload the user
const user2 = await orm.em.findOne(User, 1);
console.log(user2.firstName); // null
```

I think I can pretty easily fix this, forcing null values to be undefined when hydrating the entity, but that could be considered a breaking change, so I will include that in v3 and keep this behaviour for v2.

**Comment by B4nan** [comment]
as promised, published in v2.7.8
